### DISCUSSION TITLE
Completing the structural connectivity tutorial

### DISCUSSION THREAD

**Discussion by Joanone** [neurodesk/neurodesk.github.io#107] [Q&A] [answered]
Hi everyone!

I'm almost done with my contribution on the structural connectivity (dMRI) tutorial. However i have a question:

Context:
The last stage of the script (constructing the connectome) requires two input files:
-fs_default.txt
-freesurferColorLUT.txt

During my analysis, i copied the fs_default.txt file  and freesurferColorLUT.txt file manually to my subject folders from the mrtrix3 software and $FREESURFERHOME  outside neurodesk. 

Question:
 What is the best way future users can get access to these files without leaving the confines of neurodesk?

**Comment by stebo85** [comment, marked as answer]
Dear @Joanone,

That's fantastic to hear :) Looking forward to see the tutorial!

I am not sure I understood the question 100%, but what I understood is: Are fs_default.txt and freesurferColorLUT.txt inside Neurodesk somewhere so you can use them in the mrtrix container?

If that was the question, then: Yes, they are in the freesurfer container and the mrtrix container :)
fs_default.txt is in the mrtrix3 container under: /opt/mrtrix3-3.0.3/share/mrtrix3/labelconvert/fs_default.txt 
FreeSurferColorLUT.txt is in the freesurfer container under: /opt/freesurfer-7.2.0/FreeSurferColorLUT.txt

You can copy these files from there to a central directory like /neurodesktop-storage/

Cheers
Steffen

**Reply by Joanone** [reply]
Okay Steff. Thanks. i'll check this out.

**Reply by Joanone** [reply]
Hello Steff, tried copying the files via the main terminal,  to an external storage i mounted earlier. It didn't work.

![e85cc9ca4d2151ca3f2c4d98e2d5057](https://user-images.githubusercontent.com/83198624/161000269-82b0bab6-b7a8-40a7-a35a-6b896f05780c.png)

However, when i opened the individual singularity containers and copied them, it worked. Just wanted to confirm if there's a better way of copying the two files via one terminal so that its more convenient. 

![cdec72c877025d3118dbdde7d133472](https://user-images.githubusercontent.com/83198624/161000533-bc2c9c8c-5019-4f20-84dc-5bc1f98d814c.png)

![a82b3faecdb5dc80dac667082259bd0](https://user-images.githubusercontent.com/83198624/161000582-23add46c-3811-4fea-a476-b06b2daae4c3.png)


Thanks.
Joan

**Reply by stebo85** [reply]
Dear @Joanone,

Yes, that all makes sense :)

Opening the mrtrix and freesurfer containers and copying the files from there is the best way of doing it (and it will work under all circumstances). The applications are all isolated from each other for enhance reproducibility.

However, it is also possible to access the files from the main terminal - but this is not guaranteed to work in all cases. The files would then be here:

/cvmfs/neurodesk.ardc.edu.au/containers/freesurfer_7.2.0_20211130/freesurfer_7.2.0_20211130.simg/usr/local/freesurfer/7.2.0-1/FreeSurferColorLUT.txt

/cvmfs/neurodesk.ardc.edu.au/containers/mrtrix3_3.0.3_20210917/mrtrix3_3.0.3_20210917.simg/opt/mrtrix3-3.0.3/share/mrtrix3/labelconvert/fs_default.txt

All the best
Steffen

**Reply by Joanone** [reply]
Thanks a lot! @stebo85

**Comment by Joanone** [comment]
I have been trying to submit the tutorial and open a pull request so that it can be reviewed but i'm currently stuck at step 5 in the workflow template stated here:

https://www.neurodesk.org/tutorials/documentation/workflowtemplate/

Checking that the pages point to gh-pages, got an error stating that: 

Domain is not a valid public domain.

Domain is not properly formatted.

Thanks
Joan

**Reply by aswinnarayanan** [reply]
Hi @Joanone. Are you able to "Fetch Upstream > Fetch and Merge" the latest changes from the upstream branch?

**Reply by Joanone** [reply]
Hi @aswinnarayanan , yes, just did that now.

**Reply by Joanone** [reply]
Thanks. I have been able to create the pull request now.
